loader/common/tree_loader.py

```diff
@@ -225,20 +226,33 @@ def _get_max_height_from_node(self, tree, node):
 
 
 
-    def _check_if_child_is_single_internal_node(self, tree, curr_children, todo_list, curr_node, ordering):
+    def _merge_if_child_is_single_internal_node(self, curr_node, ordering):
+        curr_children = ordering[curr_node]
+
         # if only one child
-        if len(curr_children) == 1:    
-            # if child is a leaf node
-            if self._get_max_height_from_node(tree, curr_children[0]) == 0:
-                return (curr_children, todo_list)
-            # if child is an internal node
-            else:
-                todo_list = [[child, curr_node] for child in curr_children] + todo_list
-                curr_node = todo_list.pop(0)[0]
-                curr_children = ordering[curr_node]
-                return self._check_if_child_is_single_internal_node(tree, curr_children, todo_list, curr_node, ordering)
+        if len(curr_children) == 1:
+            try:
+                # check if child is a leaf node
+                curr_gchildren = ordering[curr_children[0]]
+                curr_node = curr_children[0]
+                return self._merge_if_child_is_single_internal_node(curr_node, ordering)
```

Comment:
Doesn't look like any merging actually happens? You just change the pointer and go to the next thing. You should update both curr_node and ordering, then recurse.
```suggestion
                curr_node = curr_node + "," curr_children[0]
                ordering[curr_node] = curr_gchildren
                return self._merge_if_child_is_single_internal_node(curr_node, ordering)
```

Reply:
I don't think ordering is ever supposed to be modified. It's my understanding that ordering is a static dictionary where the keys are nodes and their corresponding values are their children. Ordering always needs to be passed in with the function in order (haha) to get the children of the node that is currently being examined.

Reply:
I'm not entirely sure what line 238 is doing... was your thought to merge strings? If so, won't that mess up the look up the next time the function is called? Ie. line 233
```
curr_children = ordering[curr_node]
```

Reply:
The way you're doing it completely erases the original curr_node, which is not the functionality that we want. We still need to preserve all the original information, which is why the merging needs to happen. So opt for merging IDs (line 238), and that's why you need to also update ordering to let it know that this new merged ID points to this specific set of children (line 239).